connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/SingleFieldPath.java

```diff
@@ -0,0 +1,517 @@
+
+        Schema current = schema;
+        for (String pathSegment : stepsWithoutLast()) {
+            final Field field = current.field(pathSegment);
+            if (field != null) {
+                current = field.schema();
+            } else {
+                return null;
+            }
+        }
+        return current.field(lastStep());
+    }
+
+    /**
+     * Access a value at the current path within a schema-based {@code Struct}
+     * If object is not found, then {@code null} is returned.
+     */
+    public Object valueFrom(Struct struct) {
+        if (struct == null) return null;
+
+        Struct current = struct;
+        for (String pathSegment : stepsWithoutLast()) {
+            // Check to see if the field actually exists
+            if (current.schema().field(pathSegment) == null) {
+                return null;
+            }
+            current = current.getStruct(pathSegment);
+        }
+
+        if (current.schema().field(lastStep()) != null) {
+            return current.get(lastStep());
+        } else {
+            return null;
+        }
+    }
+
+    List<String> stepsWithoutLast() {
+        return steps.subList(0, lastStepIndex());
+    }
+
+    /**
+     * Access a value at the current path within a schemaless {@code Map<String, Object>}.
+     * If object is not found, then {@code null} is returned.
+     */
+    @SuppressWarnings("unchecked")
+    public Object valueFrom(Map<String, Object> map) {
+        if (map == null) return null;
+
+        if (steps.size() == 1) {
+            return map.get(steps.get(0));
+        } else {
+            Map<String, Object> current = map;
+            for (int i = 0; i < steps.size(); i++) {
+                if (current == null) {
+                    return null;
+                }
+                if (i == lastStepIndex()) {
+                    return current.get(steps.get(i));
+                } else {
+                    current = (Map<String, Object>) current.get(steps.get(i));
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Access {@code Map} fields and apply functions to update field values.
+     *
+     * <p>If path is not found, no function is applied, and the path is ignored.
+     *
+     * <p>Other fields keep values from original struct.
+     *
+     * @param originalValue schema-based data value
+     * @param whenFound     function to apply when current path(s) is/are found
+     * @return updated data value
+     */
+    public Map<String, Object> updateValueFrom(
+        Map<String, Object> originalValue,
+        MapValueUpdater whenFound
+    ) {
+        return updateValue(originalValue, 0, whenFound,
+            (originalParent, updatedParent, fieldPath, fieldName) -> {
+                // filter out
+            },
+            (originalParent, updatedParent, fieldPath, fieldName) ->
+                updatedParent.put(fieldName, originalParent.get(fieldName)));
+    }
```

Comment:
This isn't used in non-testing code; can we remove it for now and add it back in a follow-up if it's necessary for other SMTs?